src/query/graphite/native/builtin_functions.go

```diff
@@ -1591,16 +1591,65 @@ func changed(ctx *common.Context, seriesList singlePathSpec) (ts.SeriesList, err
 	})
 }
 
-// movingMedian takes one metric or a wildcard seriesList followed by a a quoted string
-// with a length of time like '1hour' or '5min'. Graphs the median of the preceding
-// datapoints for each point on the graph. All previous datapoints are set to None at
-// the beginning of the graph.
-func movingMedian(ctx *common.Context, _ singlePathSpec, windowSize string) (*binaryContextShifter, error) {
-	interval, err := common.ParseInterval(windowSize)
+// windowPointsLength calculates the number of window points in a interval
+func windowPointsLength(series *ts.Series, interval time.Duration) int {
+	return int(interval / (time.Duration(series.MillisPerStep()) * time.Millisecond))
+}
+
+type movingImplementationFn func(window []float64, values *ts.MutableValues, windowPoints int, i int)
```

Comment:
`ts.MutableValues` is already an interface and hence a pointer type, no need to make `*ts.MutableValues`.

Just change to `vals ts.MutableValues` and just use `vals.SetValueAt(...)`, this will work fine since because it's an interface and a reference underneath it will correctly pass by ref and modify the underlying struct by ref.